tfrecord/iterator_utils.py

```diff
@@ -1,37 +1,74 @@
-import typing
+"""Iterator utils."""
+
+import warnings
+from typing import Any, Callable, Iterable, Iterator, List
```

Comment:
import typing.
I prefer not to pollute the namespace.
Do this for the other places as well, import modules not types/classes/objects.